packages/grant-explorer/src/features/categories/CategoryCard.tsx

```diff
@@ -12,8 +12,17 @@ const CategoryCard = ({ category, isLoading }: CategoryCardProps) => {
     return <div>...</div>;
   }
   const { id, name, projects } = category;
+  const handleCardClick = () => {
+    // Pushing event to the data layer
+    window.dataLayer = window.dataLayer || [];
+    window.dataLayer.push({
+      event: "category_card_click",
+      categoryId: id,
+      categoryName: name,
+    });
+  };
```

Comment:
I see this is a draft PR but just wanted to drop a suggestion to move this into a hook:
```tsx
function useTrackEvent() {
  return (event: string, params: Record<string, string | number>) => {
    window.dataLayer = window.dataLayer || [];
    window.dataLayer.push({ event, ...params });
  }
}
```

Another idea is to create a component we can wrap others in:

```tsx
const TrackEvent = ({ children, event, params, ...props }) => {
  const track = useTrackEvent();
  return createElement(children, { ...props, onClick: track(event, params) });
};

<TrackEvent event="collection_card_click" params={{ collectionId: id, collectionName: name }}>
  <BasicCard ...>
    ...
  </BasicCard>
</TrackEvent>
```